src/pages/Checkout/CheckoutPage.vue

```diff
@@ -875,6 +876,9 @@ export default {
 				if (this.checkingOutAsGuest) {
 					checkoutAdditionalQueryParams += `&optedIn=${this.userOptedIn}`;
 				}
+				if (payload.username) {
```

Comment:
Given the check above for whether an object, might be good to have optional chaining here just in case.